src/main/java/jenkins/scm/api/SCMFile.java

```diff
@@ -78,6 +78,7 @@ protected SCMFile() {
      * Constructor for any entry that is not the root.
      *
      * @param parent the parent reference or {@code null} if this is the root object.
+     * @param name   the name of this entry.
```

Comment:
My github implementation creates the intermediates without making remote calls, only making remote calls for those intermediates that get checked.

If you are really objecting I can move some of this stuff to a Base class in `jenkins.scm.impl` but essentially I see all implementations being simpler with this than the other way